executor.go

```diff
@@ -62,7 +62,16 @@ func NewExecutor() *Executor {
 }
 
 // Execute executes a PQL query.
-func (e *Executor) Execute(ctx context.Context, index string, q *pql.Query, slices []uint64, opt *ExecOptions) ([]interface{}, error) {
+func (e *Executor) Execute(ctx context.Context, index string, q *pql.Query, slices []uint64, opt *ExecOptions) (results []interface{}, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			if r.(error).Error() == "runtime error: makeslice: len out of range" {
```

Comment:
better to be too specific than too general I think. We wouldn't want to report the wrong thing back.